Gems/GradientSignal/Code/Source/Editor/EditorImageGradientComponentMode.cpp

```diff
@@ -61,26 +60,31 @@ namespace GradientSignal
 
         //! Get the original gradient value for the given pixel index.
         //! Since we "lazy-cache" our unmodified image as tiles, create it here the first time we request a pixel from a tile.
-        float GetOriginalPixelValue(const PixelIndex& pixelIndex)
+        AZStd::pair<float, float> GetOriginalPixelValueAndOpacity(const PixelIndex& pixelIndex)
         {
             uint32_t tileIndex = GetTileIndex(pixelIndex);
             uint32_t pixelTileIndex = GetPixelTileIndex(pixelIndex);
 
             // Create the tile if it doesn't already exist.
             CreateImageTile(tileIndex);
 
-            return m_paintedImageTiles[tileIndex]->first[pixelTileIndex];
+            return
+            {
+                m_paintedImageTiles[tileIndex]->m_unmodifiedData[pixelTileIndex],
+                m_paintedImageTiles[tileIndex]->m_modifiedDataOpacity[pixelTileIndex]
+            };
         }
 
         //! Set a modified gradient value for the given pixel index.
-        void SetModifiedPixelValue(const PixelIndex& pixelIndex, float modifiedValue)
+        void SetModifiedPixelValue(const PixelIndex& pixelIndex, float modifiedValue, float opacity)
         {
             uint32_t tileIndex = GetTileIndex(pixelIndex);
             uint32_t pixelTileIndex = GetPixelTileIndex(pixelIndex);
 
             AZ_Assert(m_paintedImageTiles[tileIndex], "Cached image tile hasn't been created yet!");
 
-            m_paintedImageTiles[tileIndex]->second[pixelTileIndex] = modifiedValue;
+            m_paintedImageTiles[tileIndex]->m_modifiedData[pixelTileIndex] = modifiedValue;
+            m_paintedImageTiles[tileIndex]->m_modifiedDataOpacity[pixelTileIndex] = opacity;
         }
```

Comment:
nit: I'd do tile creation on set, not get. get could just return defaults if the tile didn't exist.

Reply:
I should probably do lazy tile creates on _both_ for safety, but it needs to at least be on Get() for the code to work. The "unmodifiedData" in the tile is the cached original gradient values. If Get() just returned default values instead of the gradient values, then painting would erase the gradient instead of blend with it.